compiler/src/pipeline.rs

```diff
@@ -108,6 +111,21 @@ struct Arguments<T: FieldElement> {
     backend: Option<BackendType>,
 }
 
+impl<T: FieldElement> Clone for Arguments<T> {
+    fn clone(&self) -> Self {
+        assert!(
```

Comment:
I don't think this is a great solution, because it wouldn't allow the user to specify a callback before the cloning happens. In the use-case I have in mind though (branching out starting from witgen for continuations), it would work.

Does anyone have a good solution?